GitCommands/PathUtil.cs

```diff
@@ -72,20 +77,20 @@ public static bool IsLocalFile(string fileName)
         }
 
         /// <summary>
-        /// A naive way to check whether the given path is a URL by checking
-        /// whether it starts with either 'http', 'ssh' or 'git'.
+        /// A slightly less naive way to check whether the given path is a URL by checking
+        /// whether it matches a regular expression defined in <see cref="UrlRegex"/>.
         /// </summary>
+        /// <remarks>
+        /// Certain strings such as "http://" or "ssh:" are not considered valid URLs
+        /// as they don't have a host, port or path.
+        /// </remarks>
         /// <param name="path">A path to check.</param>
         /// <returns><see langword="true"/> if the given path starts with 'http', 'ssh' or 'git'; otherwise <see langword="false"/>.</returns>
         [Pure]
         public static bool IsUrl(string path)
         {
             return !string.IsNullOrEmpty(path)
-                && (path.StartsWith("http:", StringComparison.CurrentCultureIgnoreCase)
-                 || path.StartsWith("https:", StringComparison.CurrentCultureIgnoreCase)
-                 || path.StartsWith("git:", StringComparison.CurrentCultureIgnoreCase)
-                 || path.StartsWith("ssh:", StringComparison.CurrentCultureIgnoreCase)
-                 || path.StartsWith("file:", StringComparison.CurrentCultureIgnoreCase));
+                && UrlRegex.IsMatch(path);
```

Comment:
RegEx parsing of this kind of thing makes me a bit uneasy, as the format is generally quite involved. Why not use `Url.TryParse` instead?

Reply:
I'm not sure if there is a Url.TryParse method (unless I'm looking in the wrong place?), but is this what you're referring to?
[System.UriParser.IsWellFormedOriginalString](https://docs.microsoft.com/en-us/dotnet/api/system.uriparser.iswellformedoriginalstring)

Reply:
Ah, yes, `Uri.TryCreate` seems like a better match: https://docs.microsoft.com/en-us/dotnet/api/system.uri.trycreate?view=net-5.0#System_Uri_TryCreate_System_String_System_UriKind_System_Uri__

Reply:
I've tried using Uri.TryCreate and it returns false if the string has any characters before or after the URL, which was the original problem - someone had a string in clipboard that of the form "git clone https://...." that wasn't getting detected as a valid potential URL because of the "git clone" text before the URL. I feel that using the regex to detect if the string contains a valid URL allows the user to manually remove non-URL text without having GE guess and potentially get it wrong.

Reply:
Given a URL cannot contain spaces, you could take the input string, split it by spaces and check each value.

My concern is that the full URL format is complex, and this regex will not cover all valid cases. The Uri class is far more likely to be correct.

Reply:
Thanks for the feedback, I've switched to using `Uri.IsWellFormedUriString` as that allows file paths to not be treated as URIs (`Uri.TryCreate` would consider them as URIs resulting in LocalRepositoryManager methods throwing an exception)